src/plugins/fs.c

```diff
@@ -817,3 +962,313 @@ gboolean bd_fs_xfs_resize (const gchar *mpoint, guint64 new_size, const BDExtraA
     g_free (size_str);
     return ret;
 }
+
+/**
+ * bd_fs_vfat_mkfs:
+ * @device: the device to create a new vfat fs on
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the creation (right now
+ *                                                 passed to the 'mkfs.vfat' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether a new vfat fs was successfully created on @device or not
+ */
+gboolean bd_fs_vfat_mkfs (const gchar *device, const BDExtraArg **extra, GError **error) {
+    const gchar *args[3] = {"mkfs.vfat", device, NULL};
+
+    return bd_utils_exec_and_report_error (args, extra, error);
+}
+
+/**
+ * bd_fs_vfat_wipe:
+ * @device: the device to wipe an vfat signature from
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether an vfat signature was successfully wiped from the @device or
+ *          not
+ */
+gboolean bd_fs_vfat_wipe (const gchar *device, GError **error) {
+    return wipe_fs (device, "vfat", TRUE, error);
+}
+
+/**
+ * bd_fs_vfat_check:
+ * @device: the device containing the file system to check
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the repair (right now
+ *                                                 passed to the 'fsck.vfat' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether an vfat file system on the @device is clean or not
+ *
+ * Note: if the file system is mounted it may be reported as unclean even if
+ *       everything is okay and there are just some pending/in-progress writes
```

Comment:
Is this really true for *vfat* or is it just copied over from the ``ext4_check`` function?